pir/cpp/client.cpp

```diff
@@ -68,42 +70,59 @@ StatusOr<Request> PIRClient::CreateRequest(std::size_t desired_index) const {
   const auto& plain_mod =
       context_->Parameters()->GetEncryptionParams().plain_modulus();
 
-  int index = desired_index;
-  vector<Ciphertext> query(DBSize() / poly_modulus_degree + 1);
-  for (size_t i = 0; i < query.size(); ++i) {
+  auto dims = context_->Parameters()->Dimensions();
+  auto indices = PIRDatabase::calculate_indices(dims, desired_index);
+
+  const size_t dim_sum =
+      std::accumulate(dims.begin(), dims.end(), decltype(dims)::value_type(0));
+
+  size_t offset = 0;
+  vector<Ciphertext> query(dim_sum / poly_modulus_degree + 1);
```

Comment:
When dim_sum = poly_modulus_degree, this should probably still be equal to 1 shouldn't it?